File: js/Neetcode_150/5_Sliding_Window/29_longest_ss_without_repeating_char.js
//Longest Substring without Repeating Char: Given a string s, find the length of the longest substring without repeating characters.

var lengthOfLongestSubstring = function(s) {
    let l=0, ans=0
    const st = new Set() 
    for(let r=0; r<s.length; r++){
        let x = s[r]
        if(!st.has(x)){
            st.add(x)
        }
        else{
            while(s[l] != x){
                st.delete(s[l])
                l++
            }
            l++
        }
        ans = Math.max(ans, r-l+1)
    }
    return ans
};

File: c++/Neetcode_150/5_Sliding_Window/29_longest_ss_without_repeating_char.cpp
//Longest Substring without Repeating Char: Given a string s, find the length of the longest substring without repeating characters.

int lengthOfLongestSubstring(string s) {
    unordered_set<char> st;
    int ans = 0, l=0;
    for(int r=0; r<s.size(); r++){
        char x = s[r];
        if(st.find(x) == st.end()){
            st.insert(x);
        }
        else{
            while(s[l] != x){
                st.erase(s[l]);
                l++;
            }
            l++;
        }
        ans = max(ans, r-l+1);
    }
    return ans;
}